contracts/nft/src/lib.rs

```diff
@@ -0,0 +1,322 @@
+        NftAction::Mint { to, token_metadata } => nft.mint(&to, token_metadata),
+        NftAction::Burn { token_id } => nft.burn(token_id),
+        NftAction::Transfer { to, token_id } => nft.transfer(&to, token_id),
+        NftAction::Approve { to, token_id } => nft.approve(&to, token_id),
+        NftAction::Owner { token_id } => nft.owner(token_id),
+        NftAction::IsApproved { to, token_id } => nft.is_approved_to(&to, token_id),
+    };
+    msg::reply(result, 0).expect("Failed to encode or reply with `NftEvent`.");
+}
+
+#[no_mangle]
+extern fn state() {
+    let nft = unsafe { NFT.take().expect("Unexpected error in taking state") };
+    let query: StateQuery = msg::load().expect("Unable to load the state query");
+    match query {
+        StateQuery::All => {
+            msg::reply(StateReply::All(nft.into()), 0).expect("Unable to share the state");
+        }
+        StateQuery::Config => {
+            msg::reply(StateReply::Config(nft.config), 0).expect("Unable to share the state");
+        }
+        StateQuery::Collection => {
+            msg::reply(StateReply::Collection(nft.collection), 0)
+                .expect("Unable to share the state");
+        }
+        StateQuery::Owner => {
+            msg::reply(StateReply::Owner(nft.owner), 0).expect("Unable to share the state");
+        }
+        StateQuery::CurrentTokenId => {
+            msg::reply(StateReply::CurrentTokenId(nft.token_id), 0)
+                .expect("Unable to share the state");
+        }
+        StateQuery::OwnerById { token_id } => {
+            msg::reply(
+                StateReply::OwnerById(nft.owner_by_id.get(&token_id).cloned()),
+                0,
+            )
+            .expect("Unable to share the state");
+        }
+        StateQuery::TokenApprovals { token_id } => {
+            let approval = nft.token_approvals.get(&token_id).cloned();
+            msg::reply(StateReply::TokenApprovals(approval), 0).expect("Unable to share the state");
+        }
+        StateQuery::TokenMetadata { token_id } => {
+            msg::reply(
+                StateReply::TokenMetadata(nft.token_metadata_by_id.get(&token_id).cloned()),
+                0,
+            )
+            .expect("Unable to share the state");
+        }
+        StateQuery::OwnerTokens { owner } => {
+            let tokens = nft
+                .tokens_for_owner
+                .get(&owner)
+                .map(|hashset| hashset.iter().cloned().collect());
+            msg::reply(StateReply::OwnerTokens(tokens), 0).expect("Unable to share the state");
+        }
+    }
+}
+
+impl From<Nft> for State {
+    fn from(value: Nft) -> Self {
+        let Nft {
+            owner_by_id,
+            token_approvals,
+            token_metadata_by_id,
+            tokens_for_owner,
+            token_id,
+            owner,
+            collection,
+            config,
+        } = value;
+
+        let owner_by_id = owner_by_id
+            .iter()
```

Comment:
Again, why not `into_iter`?
```suggestion
            .into_iter()
```